### PR TITLE
Add Lion optimizer

### LINKED ISSUE #5740: Add Lion optimizer to burn-optim
## Summary\n\nAdd the Lion (EvoLved Sign Momentum) optimizer to burn-optim.\n\nPaper: https://arxiv.org/abs/2302.06675\n\nLion keeps one exponential-moving-average tensor instead of the two moments used by AdamW. For gradient g_t, previous momentum m_(t-1), learning rate η, and decoupled weight decay λ, the update is:\n\n    c_t = β₁ m_(t-1) + (1 - β₁) g_t\n    θ_t = (1 - ηλ) θ_(t-1) - η sign(c_t)\n    m_t = β₂ m_(t-1) + (1 - β₂) g_t\n\nThe proposed defaults are β₁ = 0.9 and β₂ = 0.99, matching the paper and reference implementation.\n\n## Implementation\n\n- Add LionConfig, Lion, and serializable LionState.\n- Support decoupled weight decay, gradient clipping, parameter groups, and device migration.\n- Export the optimizer through burn-optim.\n- Test exact multi-step updates, state save/load, a tiny deterministic training smoke test, and the one-moment state-size advantage over AdamW.\n\nThe smoke tests will stay deliberately small; they validate the implementation and paper-derived properties without attempting to reproduce the paper’s large ImageNet/JFT experiments.

### PR DESCRIPTION
**Pull request by Mikyx-1**
Closes #5740.\n\nAdds the Lion optimizer with decoupled weight decay, gradient clipping, parameter-group support, device migration, and serializable state.\n\nTests:\n- burn-optim: 112 passed\n- Lion paper-smoke tests: passed\n- cargo fmt: passed\n- cargo clippy with warnings denied: passed

### CODE REVIEW

File: crates/burn-optim/src/optim/lion.rs [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        let input = Tensor::<2>::random([2, 4], Distribution::Default, &device);
+        let mut optimizer = LionConfig::new().init();
+        let grads = GradientsParams::from_grads(linear.forward(input).backward(), &linear);
+        let _linear = optimizer.step(1e-4, linear, grads);
+
+        let state_before = optimizer.to_record();
+        let bytes = optimizer.into_bytes().unwrap();
+        let optimizer = LionConfig::new().init().from_bytes(bytes).unwrap();
+
+        assert_eq!(state_before.len(), optimizer.to_record().len());
```
Comment (laggui) [inline crates/burn-optim/src/optim/lion.rs:180]: This should be extended to compare momentum values, and could additionally verify that the next step matches uninterrupted training.

This is a pre-existing pattern across optimizer state restoration tests though, so not a blocker for this PR.

**Review by laggui — APPROVED** [review]
The implementation looks correct, thanks!
